lib/tapioca/compilers/dynamic_mixin_compiler.rb

```diff
@@ -174,7 +174,7 @@ def compile_class_attributes(tree)
 
   sig { params(tree: RBI::Tree).returns([T::Array[Module], T::Array[Module]]) }
   def compile_mixes_in_class_methods(tree)
-    includes = dynamic_includes.select { |mod| (name = name_of(mod)) && !name.start_with?("T::") }
+    includes = dynamic_includes.select { |mod| (name = name_of(mod)) && !name.start_with?(/T\:\:(?!Props)/) }
```

Comment:
This looks good, but can we extract this logic out to a method and make the check more explicit please? I find the regexp highly unreadable myself. I'd rather see a `filtered_mixin?(mixin_name)` method like:
```ruby
def filtered_mixin?(mixin_name)
  if mixin_name.start_with?("T::Props")
    false # don't filter T::Props and subconstants since they have semantic value
  elsif mixin_name.start_with?("T::")
    true # rest of T:: namespace is mostly private constants
  else
    false # nothing else is filtered
  end
end
```
where we can add the reason why things are filtered or not, as well.